aten/src/ATen/native/TensorIndexing.cpp

```diff
@@ -0,0 +1,159 @@
+#include <ATen/native/TensorIndexing.h>
+
+#include <c10/util/Exception.h>
+
+namespace at {
+namespace indexing {
+
+const EllipsisIndexType Ellipsis = EllipsisIndexType();
+
+Slice::Slice() {}
+Slice::Slice(int64_t start, int64_t stop, int64_t step) : start_(start), stop_(stop), step_(step) {}
+
+const int64_t& Slice::start() const {
+  return start_;
+}
+
+const int64_t& Slice::stop() const {
+  return stop_;
+}
+
+const int64_t& Slice::step() const {
+  return step_;
+}
+
+std::ostream& operator<<(std::ostream& stream, const Slice& slice) {
+  stream << "{" << slice.start() << ", " << slice.stop() << ", " << slice.step() << "}";
```

Comment:
I am wondering why you stringify slices as the initializer list, as opposed to the Python syntax (e.g., `2::3` versus `{2, None, 3}`. It seems like it would be helpful to be able to report a slice in the Python syntax.